components/core/src/package/target.rs

```diff
@@ -254,7 +269,7 @@ supported_package_targets! {
     /// [x]: https://en.wikipedia.org/wiki/X86
     /// [isa]: https://en.wikipedia.org/wiki/Instruction_set_architecture
     /// [x86_64]: https://en.wikipedia.org/wiki/X86-64
-    ("x86_64-darwin", X86_64_Darwin, X86_64_DARWIN, "x86_64", "macos");
+    ("x86_64-darwin", X86_64_Darwin, X86_64_DARWIN, "x86_64", "macos", cfg(not(feature = "")));
```

Comment:
This `cfg(not(feature = ""))` stuff is kind of weird. I wonder if it would be necessary if we took a slightly different approach. Rather than a general `unsupported_targets` feature, would we have specific features for the various targets that we don't support by default? Or, have every target be an opt-in feature, but make our default supported targets part of the default feature set.

Also, I wonder how much added complexity the significant macroification of components/core/src/package/target.rs is saving us from. I didn't do anything about it before because we didn't need to change it. But if we're making modifications I'm inclined to at least consider a move away from that.

Reply:
Completely agree with both of these points. We could even have family of targets (eg `arm_targets`) that could enable all the arm targets, but in reality, I think the macro should just be removed which simplifies a lot of this.

Reply:
After polishing this PR more, I decided to leave the macro. It does remove some duplication and does group all of the related settings in the code. By leaving it, it also greatly reduced the complexity of this PR. If there is a consensus that it should be removed, I would be happy to tackle that but maybe as a different PR.